app/ui.js

```diff
@@ -367,14 +367,41 @@ var UI;
                 .addEventListener('click', UI.clipboardClear);
         },
 
+        // Add a call to save settings when the element changes,
+        // unless the optional parameter changeFunc is used instead.
+        addSettingChangeHandler: function(name, changeFunc) {
+            var settingElem = document.getElementById("noVNC_setting_" + name);
+            if (changeFunc === undefined) {
+                settingElem.addEventListener('change', function () {
+                    UI.saveSetting(name);
+                });
+            } else {
+                settingElem.addEventListener('change', changeFunc);
+            }
+        },
+
         addSettingsHandlers: function() {
             document.getElementById("noVNC_settings_button")
                 .addEventListener('click', UI.toggleSettingsPanel);
-            document.getElementById("noVNC_settings_apply")
-                .addEventListener('click', UI.settingsApply);
 
-            document.getElementById("noVNC_setting_resize")
-                .addEventListener('change', UI.enableDisableViewClip);
+            UI.addSettingChangeHandler('encrypt');
+            UI.addSettingChangeHandler('true_color');
+            if (Util.browserSupportsCursorURIs()) {
+                UI.addSettingChangeHandler('cursor');
```

Comment:
mostly just did the same thing as the old code here, but you are right, it's unnecessary.